frontend/ui/widget/touchmenu.lua

```diff
@@ -744,6 +746,105 @@ function TouchMenu:updateItems()
     end)
 end
 
+local MAX_MENU_DEPTH = 20 -- currently our menu needs at least 12 here
+function TouchMenu:_recurse(val, path, text, search_for, depth)
+    depth = depth + 1
+    if depth > MAX_MENU_DEPTH then
+        return
+    end
+    for i,v in pairs(val) do
+        local menu_text = nil
+
+        if type(v) == "table" and (type(i) == "number" or i:find("item")) then
+            local menu_entry = val[i].text or (val[i].text_func and  val[i].text_func())
+            local next_text = menu_entry and (text .. "→ " .. tostring(menu_entry)) or text -- drop sub_item_table
+            TouchMenu:_recurse(val[i], path .. "." .. i, next_text, search_for, depth)
```

Comment:
Any reason to use `TouchMenu:_recurse` and `TouchMenu.foundMenuItem` (method/property of the class) and not `self:_recurse` and `self.foundMenuItem` (method/property of the object/current instance) ?